patroni/dcs/etcd.py

```diff
@@ -602,9 +602,24 @@ def _load_cluster(self):
             history = nodes.get(self._HISTORY)
             history = history and TimelineHistory.from_node(history.modifiedIndex, history.value)
 
-            # get last leader operation
-            last_leader_operation = nodes.get(self._LEADER_OPTIME)
-            last_leader_operation = 0 if last_leader_operation is None else int(last_leader_operation.value)
+            # get last know leader lsn and slots
+            status = nodes.get(self._STATUS)
+            if status:
+                try:
+                    status = json.loads(status.value)
+                    last_lsn = status.get(self._OPTIME)
+                    slots = status.get('slots')
+                except Exception:
+                    slots = last_lsn = None
+            else:
+                last_lsn = nodes.get(self._LEADER_OPTIME)
+                last_lsn = last_lsn and last_lsn.value
+                slots = None
+
+            try:
+                last_lsn = int(last_lsn)
+            except Exception:
```

Comment:
It could be also ValueError...
Now I noticed that a similar block is missing for counsel: https://github.com/zalando/patroni/pull/1820/files#diff-99372e752231ac1fddf9cda076dd985f0ed8293e8ba1c06ea1d95a0530ccb959R343